src/common/commonutils/CommonUtils.h

```diff
@@ -54,7 +54,6 @@ int RenameFile(const char* original, const char* target, OsConfigLogHandle log);
 
 typedef int(*CommandCallback)(void* context);
 
-// If called from the main process thread the timeoutSeconds and callback arguments are ignored
```

Comment:
If I remember correctly, we had this for IoT Hub, we had there a quite nasty bug that led to this.

Reply:
This comment was incorrect even with the old code - timeoutSeconds and callback were never ignored, guessing the history from  the code that's why the "intermediate" process was launched.

Reply:
You cannot see the full history here. Recorded 'history' for this project is only from the time when we went public. 

There were far more versions of ExecuteCommand before that. That was the time when our primary management channel was IoT Hub and ExecuteCommand was primarily used by CommandRunner for the Digital Twins work. During that, we pass in the request to execute command on one thread, and we report the command results async and on a separate thread. The code is as-is because we had to shield in depth for all the corner cases in that scenario, where we run as an extension of the Azure PnP C SDK.